components/colorslider/index.css

```diff
@@ -38,59 +60,140 @@
     z-index: 2;
   }
 
-  &.is-disabled {
-    pointer-events: none;
-  }
+	&.is-disabled {
+		pointer-events: none;
+
+		.spectrum-ColorSlider-gradient {
+			display: none;
+		}
+	}
 }
 
 .spectrum-ColorSlider--vertical {
   display: inline-block;
 
-  width: var(--spectrum-colorslider-vertical-width);
-  height: var(--spectrum-colorslider-vertical-default-length);
+	min-block-size: var(
+		--mod-color-slider-vertical-minimum-height,
+		var(
+			--mod-color-slider-minimum-length,
+			var(--spectrum-color-slider-minimum-length)
+		)
+	);
+	block-size: var(
+		--mod-color-slider-vertical-height,
+		var(--mod-color-slider-length, var(--spectrum-color-slider-length))
+	);
+	min-inline-size: 0;
+	inline-size: var(
+		--mod-color-slider-vertical-control-track-width,
+		var(
+			--mod-color-slider-control-track-height,
+			var(--spectrum-color-control-track-width)
+		)
+	);
 
-  .spectrum-ColorSlider-handle {
-    left: 50%;
-    top: 0;
-  }
+	.spectrum-ColorSlider-handle {
+		inset-inline-start: 50%;
+		inset-block-start: 0;
+	}
 }
 
 .spectrum-ColorSlider-handle {
-  left: 0;
-  top: 50%;
-
-  &:after {
-    border-radius: var(--spectrum-colorslider-handle-hitarea-border-radius);
-    width: var(--spectrum-colorslider-handle-hitarea-width);
-    height: var(--spectrum-colorslider-handle-hitarea-height);
-  }
+	inset-inline-start: 0;
+	inset-block-start: 50%;
 }
 
 .spectrum-ColorSlider-checkerboard {
-  background:
-    repeating-conic-gradient(var(--spectrum-colorslider-checkerboard-light-color) 0% 25%, var(--spectrum-colorslider-checkerboard-dark-color) 0% 50%)
-    left top / calc(var(--spectrum-colorslider-checkerboard-size) * 2 ) calc(var(--spectrum-colorslider-checkerboard-size) * 2 );
-
-  /* the floating inset box shadow must be a separate element since <canvas> won't take it */
-  &:before {
-    content: "";
-    z-index: 1;
-    position: absolute;
-    top: 0;
-    left: 0;
-    bottom: 0;
-    right: 0;
-    border-radius: var(--spectrum-colorslider-border-radius);
-  }
+	background: repeating-conic-gradient(
+			var(
+					--mod-color-slider-checkerboard-light-color,
+					var(--spectrum-color-slider-checkerboard-light-color)
+				)
+				0% 25%,
+			var(
+					--mod-color-slider-checkerboard-dark-color,
+					var(--spectrum-color-slider-checkerboard-dark-color)
+				)
+				0% 50%
+		)
+		left top /
+		calc(
+			var(
+					--mod-color-slider-checkerboard-size,
+					var(--spectrum-color-slider-checkerboard-size)
+				) * 2
+		)
+		calc(
+			var(
+					--mod-spectrum-color-slider-checkerboard-size,
+					var(--spectrum-color-slider-checkerboard-size)
+				) * 2
+		);
+
+	--spectrum-color-slider-border-color-local: var(
+		--highcontrast-color-slider-border-color,
+		var(
+			--mod-color-slider-border-color,
+			var(--spectrum-color-slider-border-color-rgba)
+		)
+	);
+
+	/* The floating inset box shadow (border) must be a separate element since <canvas> won't take it. */
```

Comment:
Since the canvas variant was removed do we need this comment? can the code be refactored now?

Reply:
I've reworded this comment. It does look like it being semi-transparent and inset on top of the background is part of the design, so it seems like it should stay as a pseudo element, regardless of canvas. I'm keeping it as a `box-shadow` for now, though technically it could be refactored in the future to just use `border` (tried it and it threw off some z-index ordering; I'm going to err on the side of backwards compatibility by leaving this as-is for the migration).